frontend/src/components/search/SearchFilterAccordion/SearchFilterAccordion.tsx

```diff
@@ -0,0 +1,112 @@
+import SearchFilterSection from "./SearchFilterSection/SearchFilterSection";
+import SearchFilterToggleAll from "./SearchFilterToggleAll";
+import useSearchFilter from "../../../hooks/useSearchFilter";
+
+export interface AccordionItemProps {
+  title: React.ReactNode | string;
+  content: React.ReactNode;
+  expanded: boolean;
+  id: string;
+  headingLevel: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+  className?: string;
+  // handleToggle?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+export interface FilterOption {
+  id: string;
+  label: string;
+  value: string;
+  isChecked?: boolean;
+  children?: FilterOption[];
+}
+
+interface SearchFilterAccordionProps {
+  initialFilterOptions: FilterOption[];
+  title: string;
+}
+
+export function SearchFilterAccordion({
+  initialFilterOptions,
+  title,
+}: SearchFilterAccordionProps) {
+  // manage most of state in custom hook
+  const {
+    totalCheckedCount,
+    options,
+    mounted,
+    toggleOptionChecked,
+    toggleSelectAll,
+    incrementTotal,
+    decrementTotal,
+  } = useSearchFilter(initialFilterOptions);
+
+  const getAccordionTitle = () => (
+    <>
+      {title}
+      {!!totalCheckedCount && (
+        <span className="usa-tag usa-tag--big radius-pill margin-left-1">
+          {totalCheckedCount}
+        </span>
+      )}
+    </>
+  );
+
+  const getAccordionContent = () => (
+    <>
+      <SearchFilterToggleAll
+        onSelectAll={() => toggleSelectAll(true)}
+        onClearAll={() => toggleSelectAll(false)}
+      />
+      <ul className="usa-list usa-list--unstyled">
+        {options.map((option) => (
+          <li key={option.id}>
+            {/* If we have children, show a "section" dropdown, otherwise show just a checkbox */}
+            {option.children ? (
+              // SearchFilterSection will map over all children of this option
+              <SearchFilterSection
+                option={option}
+                incrementTotal={incrementTotal}
+                decrementTotal={decrementTotal}
+                mounted={mounted}
+                updateCheckedOption={toggleOptionChecked}
+                toggleSelectAll={toggleSelectAll}
+              />
+            ) : (
+              <SearchFilterCheckbox
+                option={option}
+                increment={incrementTotal}
+                decrement={decrementTotal}
+                mounted={mounted}
+                updateCheckedOption={toggleOptionChecked}
+              />
+            )}
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+
+  const accordionOptions: AccordionItemProps[] = [
+    {
+      title: getAccordionTitle(),
+      content: getAccordionContent(),
+      expanded: false,
+      id: "funding-instrument-filter",
+      headingLevel: "h4",
+    },
+  ];
+
+  return (
+    <Accordion
+      bordered={true}
+      items={accordionOptions}
+      multiselectable={true}
+      className="margin-top-4"
+    />
```

Comment:
Do we need to add the Truss component's `expanded` prop and keep that in state so that when the component redraws it stays expanded. I wonder about page loads/refreshes and whether the accordion state should be in the query parameters. cc @crystabelrangel

Reply:
It will reset on a refresh - you'd have to get into the browser's `localStorage` if you want to retain selections like that across page refresh

Reply:
Let's consider this out of scope for now. We'll revisit if it's bad UX for the accordions to be collapsed on page load.